carma-messenger-core/cpp_message/src/cpp_message.cpp

```diff
@@ -255,6 +255,19 @@ namespace cpp_message
             }
         }
         // else if SDSM to decode
+        else if(msg->message_type=="SensorDataSharingMessage"){
+            std::vector<uint8_t> array=msg->content;
+            SDSM_Message decode(this->get_node_logging_interface());
+            auto output=decode.decode_sdsm_message(array);
+            if(output)
+            {
+                sdsm_message_pub_->publish(output.value());
+            }
+            else
+            {
+                RCLCPP_WARN_STREAM( get_logger(), "Cannot decode SDSM message");
+            }
```

Comment:
This can be cleaned up a little to restrict the `output`'s scope:

```cpp
if(const auto output = decode.decode_sdsm_message(array); output)
{
    sdsm_message_pub_->publish(output.value());
} else
{
    RCLCPP_WARN_STREAM( get_logger(), "Cannot decode SDSM message");
}
```

Reply:
After internal discussion, it would be better to keep the code consistent with other occurrences.